packages/app/src/components/inspector/ColorInputFieldWithPicker.tsx

```diff
@@ -168,6 +173,26 @@ export default memo(function ColorInputFieldWithPicker({
             colors={values}
             onChangeColor={onChange}
           />
+          <Spacer.Vertical size={12} />
+          <Button
+            id={'handle-theme-color'}
+            full={true}
+            onClick={useCallback(() => {
+              const id = uuid();
+              onChange({
+                ...value,
+                swatchID: isSwatch ? undefined : id,
+              });
+              if (isSwatch) return;
+
+              const swatchName = prompt('New Swatch Name');
+              if (!swatchName) return;
```

Comment:
We should call `onChange` after we know we have a swatchName, so we don't end up in an inconsistent state